### DISCUSSION TITLE
Does Eclair save private keys on the bitcoind node?

### DISCUSSION THREAD

**Discussion by fiatjaf** [ACINQ/eclair#2138] [Q&A] [answered]
I know Eclair uses the bitcoind wallet, but does it send private keys to it? Or just uses it in watch-only mode or something like that?

**Comment by sstone** [comment, marked as answer]
That is correct: eclair does not include an onchain wallet (as opposed to c-lightning or lnd for example), and instead uses bitcoin-core to manage onchain funds:
- funding transactions are funded by bitcoin core
- closing transactions eventually return funds to bitcoin core: "transient" transactions (HTLC transactions, commitment transactions...) are handled by eclair which will create and publish spending transactions that return funds to bitcoin core when the appropriate delay (if any) has passed. Checking whether transactions have been spent/confirmed/... is done by fetching blocks and transactions

=> we don't send private keys to bitcoin core, and we also don't send public keys (to be used as watch-only addresses for example)

**Reply by fiatjaf** [reply]
What does it mean to be funded by bitcoind if bitcoind doesn't have public keys or private keys? I'm still confused.

**Comment by sstone** [comment]
It means that transactions are funded by bitcoin cores wallet through the `fundrawtransaction` RPC call (obviously the bitcoin node eclair is connected to needs to have a wallet and funds in it for this to work).

**Reply by fiatjaf** [reply]
That means bitcoind needs keys. How can it have funds if it doesn't have keys?

**Reply by sstone** [reply]
Yes, bitcoind needs to have a wallet with funds in it. I don't see how it's relevant to your question above "does eclair send private keys to the bitcoind wallet" ? We don't, we just use bitcoind's wallet.

**Reply by fiatjaf** [reply]
Sorry, I'm not familiar at all with how bitcoind wallet works. JoinMarket, for example, uses bitcoind wallet, but it keeps track of its own keys and sends pubkeys to bitcoind so the single JoinMarket seed is enough to recover everything later. I thought Eclair was doing something similar.

But what you're telling me is that Eclair just lets bitcoind generate its own keys and save them in wallet.dat files that are not recoverable in any way?

**Reply by sstone** [reply]
eclair does not track bitcoin core's wallet. it's up to users to manage their bitcoin node's funds and backup, but this is much easier now than it was 10 years ago (when you had to backup your `wallet.dat` every time it generated a new key): bitcoin core added support for deterministic wallets years ago (though they're using their own derivation scheme instead of BIP49/84/...) and more recently they implemented wallet descriptors => all  you need to do is backup the wallet file once, when it's created.

**Reply by fiatjaf** [reply]
Thank you, but I think this should definitely be written somewhere in Eclair docs.

**Reply by sstone** [reply]
Yes we'll document this (I thought we had), thanks!
